assets/js/script.js

```diff
@@ -4,124 +4,80 @@ $(document).ready(function() {
   }).then(function(html) {
     html = html.replace(/assets\/img\//g, (window.baseUrl || '') + 'assets/img/')
     $('#custombar').before(html).remove()
-    activate()
+    render()
   })
 })
 
-function activate() {
-  var el = {
-    input: $('#letters-input'),
-    letterTabs: $('.tab-letter-color-letter'),
-    letterTab1: $('#tab-letter1'),
-    letterTab2: $('#tab-letter2'),
-    lettersWrapper: $('.tab-letter-wrapper.active'),
-    swatch: $('.swatch-image-border'),
-  }
-
-  Object.keys(el).forEach(function(key) {
-    if (!el[key].length) { console.warn(`No element present for ${key}`); }
-  })
-
-  var bagContainerHeight = $(".main-bag-container").width() * .7
-  $(".main-bag-container").height(bagContainerHeight)
-  var bagContainerHeight = $(".main-letters-container").width() * .7
-  $(".main-letters-container").height(bagContainerHeight)
-  var letterTabHeight = $(".swatch-row").height() / 2
-  $(".custom-letter-container").height(letterTabHeight)
+var state = {
+  letters:      '',
+  materials:    ['white', 'white'],
+  activeLetter: 0,
+}
 
-  $(window).resize(function() {
-    var newLetterHeight = $(".letters-wrapper ").height()
-    $("#bagLetter1").css({ display: "block", height: newLetterHeight + "px" })
-    $("#bagLetter2").css({ display: "block", height: newLetterHeight + "px" })
-    bagContainerHeight = $(".main-bag-container").width() * .7
-    $(".main-bag-container").height(bagContainerHeight)
-    bagContainerHeight = $(".main-letters-container").width() * .7
-    $(".main-letters-container").height(bagContainerHeight)
-    letterTabHeight = $(".swatch-row").height() / 2
-    $(".custom-letter-container").height(letterTabHeight)
+$(document).on('input', '.js-input', function(evt) {
+  setState({
+    letters: $(this).val()
   })
+})
+$(document).on('click', '.js-swatch-link', function(evt) {
+  var materials = state.materials.concat([]);
+  materials[state.activeLetter] = $(this).data('color')
 
-  var letter1Choice = "black"
-  var letter2Choice = "black"
-
-  el.letterTabs.on('click', function() {
-    el.letterTabs.removeClass("active")
-    el.lettersWrapper.removeClass('active')
-    $(this).addClass("active")
-    $(this).parent().addClass('active')
+  setState({
+    materials: materials
   })
-
-  el.letterTab1.on('click', function() {
-    el.swatch.removeClass('active')
-    el.swatch.each(function() {
-      if ($(this).data("material-swatch") == letter1Choice) {
-        $(this).addClass("active")
-      }
-    })
+})
+$(document).on('click', '.js-tab', function(evt) {
+  setState({
+    activeLetter: $(this).data('index')
   })
+})
 
-  el.letterTab2.on('click', function() {
-    el.swatch.removeClass('active')
-    el.swatch.each(function() {
-      if ($(this).data("material-swatch") == letter2Choice) {
-        $(this).addClass("active")
-      }
-    })
-  })
+function setState(newState) {
+  state = Object.assign({}, state, newState)
```

Comment:
this is a really rudimentary implementation of `setState`. it just overwrites any keys of the old state that exist in the new state and calls `render`.